client/me/purchases/manage-purchase/notices.tsx

```diff
@@ -821,15 +848,9 @@ class PurchaseNotice extends Component {
 
 	/**
 	 * Returns an object with credit card details suitable for use as translation arguments.
-	 *
-	 * @param {Object} purchase - the purchase to get credit card details from
-	 * @returns {Object}  Translation arguments containing information on the card type, number, and expiry
 	 */
-	creditCardDetails = ( purchase ) => {
+	creditCardDetails = ( creditCard: PurchasePaymentCreditCard ) => {
```

Comment:
This was the biggest change I had to make. Previously this function required passing a purchase that was made with a credit card, but there's no type for that; purchases can be made with various payment methods. Instead I opted to change the function to require the credit card data from the purchase; this puts the onus on the caller to collect it from the purchase object and verify that it exists.

Reply:
I had to look up `PurchasePaymentCreditCard`  and now it makes sense